src/test/kotlin/com/atlassian/performance/tools/awsinfrastructure/SuppressingRunnableTest.kt

```diff
@@ -0,0 +1,69 @@
+package com.atlassian.performance.tools.awsinfrastructure
+
+import org.hamcrest.CoreMatchers.*
+import org.junit.Assert.*
+import org.junit.Test
+
+class SuppressingRunnableTest {
+    @Test
+    fun shouldExecuteAllEvenIfFirstFails() {
+        var executed1 = false
+        var executed2 = false
+        var executed3 = false
+        val runnable = SuppressingRunnable(
+            listOf(
+                Runnable { executed1 = true; throw Exception("Fail 1") },
+                Runnable { executed2 = true; throw Exception("Fail 2") },
+                Runnable { executed3 = true; throw Exception("Fail 3") }
+            )
+        )
+
+        try {
+            runnable.run()
+        } catch (e: Exception) {
+            // Expected and ignored, so that we can go to asserts
+        }
+
+        assertTrue(executed1)
+        assertTrue(executed2)
+        assertTrue(executed3)
+    }
+
+    @Test
+    fun shouldThrowAllFailures() {
+        val runnable = SuppressingRunnable(
+            listOf(
+                Runnable { throw Exception("Banana") },
+                Runnable { throw Exception("Apple") },
+                Runnable { throw Exception("Pear") },
+                Runnable { throw Exception("Peach") }
+            )
+        )
+
+        val exception = try {
```

Comment:
Nit:
```
        val exception = catchThrowable { runnable.run() }

        assertThat(exception)
            .isInstanceOf(Exception::class.java)
            .hasMessage("Banana")

        assertThat(exception?.suppressed)
            .extracting("message")
            .containsExactly("Apple", "Pear", "Peach")`

Reply:
I wanted to avoid explicit assert on what message lands in suppressed and what in the "root", because this is not important - the important fact is that all of the messages (and actually also stacktraces, but I don't know how to assert on that reliably) are available inside the single exception in some way.

I will see how to use AssertJ for that.